### PR TITLE
feat: tests

### PR DESCRIPTION
**Pull request by EugeniaRe**
Task: #7 

Screenshot:
image

- [x] Отрефакторила роутер
- [x] Написала тесты для роутов из router.ts

Additional info:
Для stack что-то у меня не получается 100%

### CODE REVIEW

File: src/router/badge/plain.ts [hunk 1]
```diff
@@ -1,12 +1,10 @@
 import { type Context } from 'jsr:@oak/oak/'
 import { badgePlain } from '~/src/badge/plain.ts'
+import { badgeProps } from '../badgeRoutes.ts'
```
Comment (mrHoft) [inline src/router/badge/plain.ts:3]:
```suggestion
import type { TBadgeProps } from '../badgeRoutes.ts'
```
Я долго не мог понять что это за переменная такая badgeProps... Ну есть же договорённость как должны именоваться типы! То же касается импортов типов.

**Review by mrHoft — APPROVED** [review]
Молодец что compilerOptions почистила, а то чуть не забыли про них!

Не критично:
- Очищай кэш после каждого теста. Объём не такой большой, моки совсем крохотные, но нужно привыкать "прибирать за собой".

Критично:
- Избавься от "as any".

### Tests strengths:
- Good Coverage: Tests multiple routes including /plain, /skill, /stack, /icons, /logs, /, /favicon, and file/image handling.
- Edge Cases: Includes tests for error conditions (400 status codes, invalid inputs, empty requests).
- Parameter Testing: Properly tests routes with and without query parameters.
- Response Type Validation: Correctly checks that SVG routes return the proper content type.
- Cache Testing: Tests caching behavior for image requests.

**Comment by EugeniaRe** [comment]
Поправила type и any. Не могу понять что именно нужно очищать? ctx?

File: src/tests/router.test.ts [hunk 2]
```diff
@@ -100,7 +101,7 @@ describe('Badge routes', () => {
       method: 'GET',
       path: '/logs',
       params: { level: 'info' },
```
Comment (mrHoft) [inline src/tests/router.test.ts:103]:
```suggestion
      path: '/logs/info',
      params: { level: 'info' },
```
Не существенно.

**Review by mrHoft — APPROVED** [review]
В этом тесте можешь не очищать моки, так как они не значительны.
Я лишь отметил что очищение является хорошей практикой при написании тестов.
